File: VB/VB/WinWebSolution.Module.Win/WinShowCustomFormWindowController.vb
Imports Microsoft.VisualBasic
Imports System
Imports WinCustomUserControlLibrary

Namespace WinWebSolution.Module.Win
	Public Class WinShowCustomFormWindowController
		Inherits ShowCustomFormWindowController
		Protected Overrides Sub ShowCustomNonModalForm()
			Dim form As New WinCustomForm()
			form.Text = GetWindowCaption()
			form.Show()
		End Sub
		Protected Overrides Sub ShowCustomModalForm()
			Using form As New WinCustomForm()
				form.Text = GetWindowCaption()
				form.ShowDialog()
			End Using
		End Sub
	End Class
End Namespace

File: CS/WinWebSolution.Module.Win/WinShowCustomFormWindowController.cs
using System;
using WinCustomUserControlLibrary;

namespace WinWebSolution.Module.Win {
    public class WinShowCustomFormWindowController : ShowCustomFormWindowController {
        protected override void ShowCustomNonModalForm() {
            WinCustomForm form = new WinCustomForm();
            form.Text = GetWindowCaption();
            form.Show();
        }
        protected override void ShowCustomModalForm() {
            using (WinCustomForm form = new WinCustomForm()) {
                form.Text = GetWindowCaption();
                form.ShowDialog();
            }
        }
    }
}
